helm-charts/common/data-prep/templates/deployment.yaml

```diff
@@ -28,8 +28,36 @@ spec:
       serviceAccountName: {{ include "data-prep.serviceAccountName" . }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: wait-for-db
+          envFrom:
+            - configMapRef:
+                name: {{ include "data-prep.fullname" . }}-config
+            {{- if .Values.global.extraEnvConfig }}
+            - configMapRef:
+                name: {{ .Values.global.extraEnvConfig }}
+                optional: true
+            {{- end }}
+          image: busybox:1.36
+          command: ["sh", "-c"]
+          args:
+            - |
+              {{- if eq "REDIS" .Values.DATAPREP_BACKEND }}
+              TESTHOST=$(REDIS_HOST);
+              TESTPORT=$(REDIS_PORT);
+              {{- else if eq "MILVUS" .Values.DATAPREP_BACKEND }}
+              TESTHOST=$(MILVUS_HOST);
+              TESTPORT=$(MILVUS_PORT);
+              {{- end }}
+              retry_count={{ .Values.retryCount | default 60 }};
```

Comment:
IMHO better name would be e.g. `retryTimeoutSeconds`, because that way there's no dependency on whether checks are done at 1s, 2s or 5s interval...

Reply:
Are you suggesting we do a **single** `nc -z -w retryTimeoutSeconds <hostname> <port>` test, **without** the while loop?

Reply:
Ok, if that is not reliable, loop can still be made independent of sleep interval:
```
timeout={{ .Values.retryTimeoutSeconds | default 60 }};
j=1
while ! nc -z ${TESTHOST} ${TESTPORT}; do
    if [[ $j -ge ${timeout} ]] && echo "ERROR: ${TESTHOST}:${TESTPORT} is NOT reachable within $j seconds!" && exit 1;
    j=$((j+2)); sleep 2;
done;
```

Reply:
I adjust the wait time code a little bit to be more time accurate as much as I could. In a very nasty network environment where a normal successful tcp 3-way handshakes takes more than 1 seconds(which is very rarely I assume), users can set `waitTimeout` to a larger value.

Reply:
Seems fine to me!